src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs

```diff
@@ -1648,5 +1649,29 @@ public override void WriteToken(XmlWriter writer, SecurityToken token)
         {
             throw new NotImplementedException();
         }
+
+        /// <inheritdoc/>
+        public override Task<TokenValidationResult> ValidateTokenAsync(string token, TokenValidationParameters validationParameters)
+        {
+            try
+            {
+                var claimsPrincipal = ValidateToken(token, validationParameters, out var validatedToken);
+                return Task.FromResult(new TokenValidationResult
+                {
+                    SecurityToken = validatedToken,
+                    ClaimsIdentity = claimsPrincipal?.Identity as ClaimsIdentity,
+                    IsValid = true,
+                });
+            }
+            // only catch and return the SecurityTokenException via the TokenValidationResult, all other types of exceptions should be handled by the caller, ex., ArgumentException, ArgumentNullException...
+            catch (SecurityTokenException ste)
```

Comment:
>catch [](http://example.com/codeflow?start=12&length=5)

Thanks Danny, this LGTM.

One concern I have (now that I actually got what I asked for) is the consistency of the devEx here. These are new methods so I don't think we'd break anyone directly but anyone taking a dependency on these might mistakenly think they operate on the same old security token model where the validate methods throw an exception in the failure condition.

They might even write some bad tests (like passing null token) and confirm that this is how it works without passing a token with , e.g. a bad aud or iss. If they do that we might be in a similar case what we faced last year where folks don't check the return value and mistakenly don't actually validate tokens.

While I agree with the direction generally for the reasons in my other comment, I'd like to get Brent's (or someone with more team history) eyes on this before approving and then pushing out a Wilson release that locks us in here.

Reply:
All token handlers ValidateTokenAsync() now catch the general exception to be consistent as discussed.